internal/bminventory/inventory.go

```diff
@@ -1841,6 +1842,13 @@ func (b *bareMetalInventory) UpdateClusterInternal(ctx context.Context, params i
 		return nil, err
 	}
 
+	if b.ocmClient != nil && b.Config.WithAMSSubscriptions && newClusterName != "" && newClusterName != cluster.Name {
```

Comment:
maybe we can move this to line 1758, instead of checking newClusterName != "" again.

Reply:
We need it to be inside the transaction so we don't need to rollback everything manually if we fail.